frontend/package.json

```diff
@@ -12,38 +12,38 @@
   },
   "private": true,
   "dependencies": {
-    "@angular/common": "^4.0.0",
-    "@angular/compiler": "^4.0.0",
-    "@angular/core": "^4.0.0",
-    "@angular/forms": "^4.0.0",
-    "@angular/http": "^4.0.0",
-    "@angular/platform-browser": "^4.0.0",
-    "@angular/platform-browser-dynamic": "^4.0.0",
-    "@angular/router": "^4.0.0",
-    "@ng-bootstrap/ng-bootstrap": "^1.0.0-beta.1",
-    "bootstrap": "^4.0.0-beta",
+    "@angular/common": "^5.0.1",
+    "@angular/compiler": "^5.0.1",
+    "@angular/core": "^5.0.1",
+    "@angular/forms": "^5.0.1",
+    "@angular/http": "^5.0.1",
+    "@angular/platform-browser": "^5.0.1",
+    "@angular/platform-browser-dynamic": "^5.0.1",
+    "@angular/router": "^5.0.1",
+    "@ng-bootstrap/ng-bootstrap": "^1.0.0-beta.5",
+    "bootstrap": "^4.0.0-beta.2",
     "core-js": "^2.4.1",
     "ngx-cookie": "^1.0.0",
-    "rxjs": "^5.1.0",
+    "rxjs": "^5.5.2",
     "zone.js": "^0.8.4"
   },
   "devDependencies": {
-    "@angular/cli": "1.0.0",
-    "@angular/compiler-cli": "^4.0.0",
-    "@types/jasmine": "2.5.38",
-    "@types/node": "~6.0.60",
-    "codelyzer": "~2.0.0",
-    "jasmine-core": "~2.5.2",
-    "jasmine-spec-reporter": "~3.2.0",
-    "karma": "~1.4.1",
-    "karma-chrome-launcher": "~2.0.0",
+    "@angular/cli": "1.5.0",
+    "@angular/compiler-cli": "^5.0.1",
+    "@types/jasmine": "2.6.3",
+    "@types/node": "~8.0.50",
+    "codelyzer": "~4.0.1",
+    "jasmine-core": "~2.8.0",
+    "jasmine-spec-reporter": "~4.2.1",
+    "karma": "~1.7.1",
+    "karma-chrome-launcher": "~2.2.0",
     "karma-cli": "~1.0.1",
     "karma-jasmine": "~1.1.0",
     "karma-jasmine-html-reporter": "^0.2.2",
-    "karma-coverage-istanbul-reporter": "^0.2.0",
-    "protractor": "~5.1.0",
-    "ts-node": "~2.0.0",
-    "tslint": "~4.5.0",
-    "typescript": "~2.2.0"
+    "karma-coverage-istanbul-reporter": "^1.3.0",
+    "protractor": "~5.2.0",
+    "ts-node": "~3.3.0",
+    "tslint": "~5.8.0",
+    "typescript": "^2.6.1"
```

Comment:
again, please double check, but I think angular5 currently requires typescript 2.4.x (it might have changed since)

Reply:
angular upgrade guide propose typescript 2.4.2: https://angular-update-guide.firebaseapp.com/
```
npm install typescript@2.4.2 --save-exact
```
so myabe we should stick to this one and reevaluate at another major angular upgrade